### PR TITLE
Get sas token to upload the zip files to storage container

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1248

### Change description ###

In this PR,  testing the blob router SAS token endpoint.
In the next PRs,
- Prepare test data 
- Upload to the storage container with SAS token
- verify response 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -149,12 +150,17 @@ ext.libraries = [
 dependencies {
   compile group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.13.3'
   compile group: 'org.apache.logging.log4j', name: 'log4j-to-slf4j', version: '2.13.3'
+  compile group: 'com.azure', name: 'azure-storage-blob', version: '12.6.1'
+  compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.11.0'
```
Comment (doncem) [inline build.gradle:154]: there are no source code actually. i suppose for tools kind of stuff we might have it but for now it's not there

Comment (aliveni) [reply build.gradle]: Update the Gradle file

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    @Test
+    public void testSasToken() throws Exception {
+        assertThat(blobRouterUrl).isNotEmpty();
+
+        Response sasTokenResponse = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(blobRouterUrl)
+            .header(HttpHeaders.CONTENT_TYPE, "application/json")
+            .when().get("/token/bulkscan")
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java:32]: missing new line I think

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Test
+    public void testSasToken() throws Exception {
+        assertThat(blobRouterUrl).isNotEmpty();
+
+        Response sasTokenResponse = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(blobRouterUrl)
+            .header(HttpHeaders.CONTENT_TYPE, "application/json")
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java:31]: there should be a constant for `"application/json"` somewhere

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+        Map<String, String[]> queryParams = StorageImplUtils.parseQueryStringSplitValues(Utility.urlDecode(sasToken));
+
+        assertThat(queryParams.get("sig")).isNotNull(); // this is a generated hash of the resource string
+        assertThat(queryParams.get("sv")).contains("2019-07-07"); // azure api version is latest
+        OffsetDateTime expiresAt = OffsetDateTime.parse(queryParams.get("se")[0]); // expiry datetime for the signature
+        assertThat(expiresAt).isBetween(now(), now().plusSeconds(300));
+        assertThat(queryParams.get("sp")).contains("cl"); // access permissions(create-c,list-l)
+        assertThat(queryParams.get("spr")).containsExactlyInAnyOrder("https", "http");
+    }
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java:52]: I think there's no need for that in end to end test

Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java:52]: just 200 OK

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java (hunk 3 above)
Comment (aliveni) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java:22]: I tried apache httpclient library but it doesn't have constant for "application/json".
In the other projects, we always used spring dependency

Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/SasTokenTest.java:22]: ok

**Pull request merged**